src/server/pkg/worker/api_server.go

```diff
@@ -385,8 +395,27 @@ func (a *APIServer) Process(ctx context.Context, req *ProcessRequest) (resp *Pro
 	// We cannot run more than one user process at once; otherwise they'd be
 	// writing to the same output directory. Acquire lock to make sure only one
 	// user process runs at a time.
-	a.Lock()
-	defer a.Unlock()
+	a.processMu.Lock()
+	defer a.processMu.Unlock()
+	// set the status for the datum
+	ctx, cancel := context.WithCancel(ctx)
+	func() {
+		a.statusMu.Lock()
+		defer a.statusMu.Unlock()
+		a.jobID = req.JobID
+		a.data = req.Data
+		a.started = time.Now()
+		a.cancel = cancel
+	}()
+	// unset the status when this function exits
+	defer func() {
+		a.statusMu.Lock()
+		defer a.statusMu.Unlock()
+		a.jobID = ""
+		a.data = nil
+		a.started = time.Time{}
+		a.cancel = nil
```

Comment:
Can we just put these four fields into a `WorkerStatus` struct or something?  The code as it is looks a little rough since we are always manipulating the four fields at once anyways.